livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py

```diff
@@ -0,0 +1,384 @@
+                    break
+                else:
+                    raise ValueError("Received unexpected data type: ", type(data))
+
+        async def recv_task():
+            nonlocal closing_ws
+            while True:
+                msg = await ws.receive()
+                if msg.type in (
+                    aiohttp.WSMsgType.CLOSED,
+                    aiohttp.WSMsgType.CLOSE,
+                    aiohttp.WSMsgType.CLOSING,
+                ):
+                    if closing_ws:  # close is expected, see SpeechStream.aclose
+                        return
+
+                    raise Exception(
+                        "AssemblyAI connection closed unexpectedly",
+                    )  # this will trigger a reconnection, see the _run loop
+
+                if msg.type != aiohttp.WSMsgType.TEXT:
+                    print("unexpected AssemblyAI message type %s", msg.type)
+                    continue
+
+                try:
+                    # received a message from AssemblyAI
+                    data = json.loads(msg.data)
+                    self._process_stream_event(data)
+                except Exception:
+                    print("failed to process AssemblyAI message")
+
+        await asyncio.gather(send_task(), recv_task())
+
+    def _end_speech(self) -> None:
+        if not self._speaking:
+            print(
+                "trying to commit final events without being in the speaking state",
+            )
+            return
+
+        if len(self._final_events) == 0:
+            return
+
+        self._speaking = False
+
+        # combine all final transcripts since the start of the speech
+        sentence = ""
+        confidence = 0.0
+        for f in self._final_events:
+            alt = f.alternatives[0]
+            sentence += f"{alt.text.strip()} "
+            confidence += alt.confidence
+
+        sentence = sentence.rstrip()
+        confidence /= len(self._final_events)  # avg. of confidence
+
+        end_event = stt.SpeechEvent(
+            type=stt.SpeechEventType.END_OF_SPEECH,
+            alternatives=[
+                stt.SpeechData(
+                    language=ENGLISH,
+                    start_time=self._final_events[0].alternatives[0].start_time,
+                    end_time=self._final_events[-1].alternatives[0].end_time,
+                    confidence=confidence,
+                    text=sentence,
+                ),
+            ],
+        )
+        self._event_queue.put_nowait(end_event)
+        self._final_events = []
+
+    def _process_stream_event(self, data: dict) -> None:
+        # see this page:
+        # https://www.assemblyai.com/docs/api-reference/streaming/realtime
+        # for more information about the different types of events
+        if data["message_type"] == "SessionBegins":
+            self._speaking = True
+            start_event = stt.SpeechEvent(type=stt.SpeechEventType.START_OF_SPEECH)
```

Comment:
Hey, since this is only being fired once, I think it is OK to not fire it at all.
I'll add a "vad" capability boolean to the STT abc in the future.
Otherwise it lgtm thanks!
We just need to add our AssemblyAI token before merging

Reply:
Hey @theomonnom ! Can you clarify what you mean? This stream event puts a start event on the event queue which signifies that a websocket connection with AssemblyAI has been established and a session has begun, not that speech itself has begun